openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -133,6 +89,307 @@ def no_workspace_dir():
+                           exc_info=True)
+            raise
+
+        return TextureResult(
+            path=destination,
+            file_hash=texture_hash,
+            colorspace=colorspace,
+            transfer_mode=COPY
+        )
+
+    @staticmethod
+    def get_extension():
+        return ".rstexbin"
+
+    @staticmethod
+    def get_redshift_tool(tool_name):
+        """Path to redshift texture processor.
+
+        On Windows it adds .exe extension if missing from tool argument.
+
+        Args:
+            tool_name (string): Tool name.
+
+        Returns:
+            str: Full path to redshift texture processor executable.
+        """
+        if "REDSHIFT_COREDATAPATH" not in os.environ:
+            raise RuntimeError("Must have Redshift available.")
+
+        redshift_tool_path = os.path.join(
+            os.environ["REDSHIFT_COREDATAPATH"],
+            "bin",
+            tool_name
+        )
+
+        return find_executable(redshift_tool_path)
+
+
+class MakeTX(TextureProcessor):
+    """Make `.tx` using `maketx` with some default settings.
+
+    Some hardcoded arguments passed to `maketx` are based on the defaults used
+    in Arnold's txManager tool.
+
+    """
+
+    def __init__(self, log=None):
+        super(MakeTX, self).__init__(log=log)
+        self.extra_args = []
+
+    def apply_settings(self, system_settings, project_settings):
+        # Allow extra maketx arguments from project settings
+        extra_args_dict = (
+            project_settings["maya"]["publish"]
+            .get("ExtractLook", {}).get("maketx_arguments", {})
+        )
+        extra_args = []
+        for flag, value in extra_args_dict.items():
+            extra_args.append(flag)
+            extra_args.append(value)
+        self.extra_args = extra_args
+
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+        """Process the texture.
+
+        This function requires the `maketx` executable to be available in an
+        OpenImageIO toolset detectable by OpenPype.
+
+        Args:
+            source (str): Path to source file.
+            colorspace (str): Colorspace of the source file.
+            color_management (dict): Maya Color management data from
+                `lib.get_color_management_preferences`
+            staging_dir (str): Output directory to write to.
+
+        Returns:
+            TextureResult: The resulting texture information.
+
+        """
+
+        maketx_path = get_oiio_tools_path("maketx")
+
+        if not maketx_path:
+            raise AssertionError(
+                "OIIO 'maketx' tool not found. Result: {}".format(maketx_path)
+            )
+
+        # Define .tx filepath in staging if source file is not .tx
+        fname, ext = os.path.splitext(os.path.basename(source))
+        if ext == ".tx":
+            # Do nothing if the source file is already a .tx file.
```

Comment:
what if we changed the colorspace though, would we expect it to regenerate the .tx?

Reply:
I'm not sure how to regenerate from a `.tx` file. In that case I think you'd regenerate from another source format, right? Is this something being done where tx files are regenerated?

Reply:
oh right yeah makes sense